pubsub/aws/aws.go

```diff
@@ -56,15 +56,16 @@ func NewPublisher(cfg SNSConfig) (pubsub.Publisher, error) {
 		if err != nil {
 			return p, err
 		}
-	} else {
-		creds = credentials.NewEnvCredentials()
 	}
 
-	p.sns = sns.New(sess, &aws.Config{
-		Credentials: creds,
-		Region:      &cfg.Region,
-		Endpoint:    cfg.EndpointURL,
-	})
+	awsConfig := &aws.Config{
+		Region:   &cfg.Region,
+		Endpoint: cfg.EndpointURL,
+	}
+	if creds != nil {
```

Comment:
If `creds` _is_ already `nil` then setting it into the `awsConfig.Credentials` as nil won't have any effect. With that in mind, I think we can drop this `if` statement and use the prior, 1 liner for `sns.New`, right?

Reply:
Yeah, I wasn't clear before, but that's what I meant: dropping the ``else`` block should be the only change needed in this PR.

Reply:
sure, done